examples/saas-clients/amazons3/working-with-amazons3-client/working_with_amazons3_client.bal

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/config;
+import ballerina/http;
+import ballerina/log;
+import ballerina/mime;
+
+import wso2/amazons3;
+
+// Constants for error codes and messages.
+const string ERROR_CODE = "Sample Error";
+const string RESPOND_ERROR_MSG = "Error in responding to client.";
+const string CLIENT_CREATION_ERROR_MSG = "Error while creating the AmazonS3 client.";
+const string BUCKET_CREATION_ERROR_MSG = "Error while creating bucket on Amazon S3.";
+const string PAYLOAD_EXTRACTION_ERROR_MSG = "Error while extracting the payload from request.";
+const string OBJECT_CREATION_ERROR_MSG = "Error while creating object on Amazon S3.";
+const string OBJECT_DELETION_ERROR_MSG = "Error while deleting object from Amazon S3.";
+const string BUCKET_DELETION_ERROR_MSG = "Error while deleting bucket from Amazon S3.";
+const string INVALID_PAYLOAD_MSG = "Invalid request payload";
+
+// Read accessKey and secretKey from config files.
+string accessKeyId = config:getAsString("ACCESS_KEY_ID");
+string secretAccessKey = config:getAsString("SECRET_ACCESS_KEY");
+
+// Create Amazons3 client configuration with the above accesskey and secretKey values.
+amazons3:ClientConfiguration amazonS3Config = {
+    accessKeyId: accessKeyId,
+    secretAccessKey: secretAccessKey
+};
+
+@http:ServiceConfig {
+    basePath: "/amazons3"
+}
+service amazonS3Service on new http:Listener(9090) {
+    @http:ResourceConfig {
+        methods: ["POST"],
+        path: "/{bucketName}"
+    }
+    // Function to create a new bucket.
+    resource function createBucket(http:Caller caller, http:Request request, string bucketName) {
+        // Create AmazonS3 client with the above amazonS3Config.
+        amazons3:AmazonS3Client|error amazonS3Client = new(amazonS3Config);     
```

Comment:
Create an AmazonS3 client out of the resource. So only one client will be created and you can use that client inside resource methods.

Reply:
Fixed. Thanks.